core/src/main/java/hudson/Functions.java

```diff
@@ -966,11 +966,13 @@ public static String getFooterURL() {
             if (d.getGlobalConfigPage()==null)  continue;
 
             if (d instanceof GlobalConfiguration) {
-                if (predicate.apply(((GlobalConfiguration)d).getCategory()))
+                if (predicate.apply((d.getCategory()))) {
                     r.add(new Tag(c.ordinal(), d));
+                }
             } else {
-                if (predicate.apply(GlobalConfigurationCategory.get(Unclassified.class)))
+                if (predicate.apply(d.getCategory())) {
                     r.add(new Tag(0, d));
```

Comment:
@amuniz Jesse left a comment on this in my original PR. Basically, this if/else is no longer needed, as every descriptor now has `getCategory()`.